### PR TITLE
Compatibility with Angular 14

### PR DESCRIPTION
**Pull request by sharmanikhil04**
…l-file-input/issues/111)

-Upgrading dependencies
-Adding stateChanges property to support changes to MatFormFieldControl in v14

### CODE REVIEW

**Review by jonaskistler — APPROVED** [review]

**Review by rRaDuCaN — APPROVED** [review]

File: libs/material-file-input/src/lib/file-input/file-input.component.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import { ControlValueAccessor, NgControl, NgForm, FormGroupDirective } from '@angular/forms';
-import { ErrorStateMatcher } from '@angular/material/core';
-import { MatFormFieldControl } from '@angular/material/form-field';
-import { FocusMonitor } from '@angular/cdk/a11y';
-import { coerceBooleanProperty } from '@angular/cdk/coercion';
+
 
 import { FileInput } from '../model/file-input.model';
 import { FileInputMixinBase } from './file-input-mixin';
+import {MatFormFieldControl} from "@angular/material/form-field";
```
Comment (JeremieLapert) [inline libs/material-file-input/src/lib/file-input/file-input.component.ts:7]: a space is missing for your import

File: libs/material-file-input/src/lib/file-input/file-input-mixin.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { FormGroupDirective, NgControl, NgForm } from '@angular/forms';
-import { ErrorStateMatcher, mixinErrorState } from '@angular/material/core';
+import {ErrorStateMatcher, mixinErrorState} from '@angular/material/core';
```
Comment (JeremieLapert) [inline libs/material-file-input/src/lib/file-input/file-input-mixin.ts:2]: a space is missing for your import

File: package.json [hunk 3]
```diff
@@ -56,24 +56,24 @@
...
+    "@angular/cli": "^14.0.1",
+    "@angular/compiler-cli": "^14.0.1",
+    "@angular/language-service": "^14.0.1",
     "@ngrx/schematics": "^13.0.1",
-    "@nrwl/cli": "13.2.1",
-    "@nrwl/eslint-plugin-nx": "13.2.1",
-    "@nrwl/linter": "13.2.1",
-    "@nrwl/tao": "13.2.1",
-    "@nrwl/workspace": "13.2.1",
+    "@nrwl/cli": "14.0.1",
```
Comment (JeremieLapert) [inline package.json:67]: We are now in version 14.3.5 for NX cf https://github.com/nrwl/nx/releases/tag/14.3.5

File: libs/material-file-input/src/lib/file-input/file-input.component.ts (hunk 1 above)
Comment (sharmanikhil04) [reply libs/material-file-input/src/lib/file-input/file-input.component.ts]: Fix this. Please review

**Review by JeremieLapert — APPROVED** [review]

**Review by JeremieLapert — APPROVED** [review]

**Review by JeremieLapert — APPROVED** [review]

**Review by rRaDuCaN — APPROVED** [review]

**Review by rRaDuCaN — APPROVED** [review]

**Review by rRaDuCaN — APPROVED** [review]

**Review by rRaDuCaN — CHANGES_REQUESTED** [review]
seems fine, just commented to trigger this darn PR

**Comment by sharmanikhil04** [comment]
> seems fine, just commented to trigger this darn PR

Hi @rRaDuCaN , I am also waiting for @merlosy  to approve and merge this PR.

**Comment by winfa** [comment]
hi @rRaDuCaN, I have already upgrade my project to angular 14, but found this issue before release. Could this issue been resolved ASAP? Or I may have to revert my code to angular 13...

**Comment by rRaDuCaN** [comment]
@sharmanikhil04 , I guess @merlosy doesn't use this account anymore, at least I don't see any activity from him. Also, it's pretty annoying that angular material doesn't provide a built in for files, the same problem with datepicker (problem solved in material ui for react to bring it up)

**Comment by winfa** [comment]
Hi @sharmanikhil04, I was trying to build based on your PR but found errors when run `npm run build` or `npm run build:lib`. Maybe the angular upgrade part may should be covered by nx.

Also, not sure if it is possible to keep the dependency as angular 13, and just update the code and keep stateChanges as optional to make this lib support angular 13 and angular 14. 

@rRaDuCaN  @sharmanikhil04 
And btw, if @merlosy does not use this account anymore. Do we have any other plans? Do we have any other person can contact with? Or maybe publish this change to anther place?

**Comment by sharmanikhil04** [comment]
> Hi @sharmanikhil04, I was trying to build based on your PR but found errors when run `npm run build` or `npm run build:lib`. Maybe the angular upgrade part may should be covered by nx.
> 
> Also, not sure if it is possible to keep the dependency as angular 13, and just update the code and keep stateChanges as optional to make this lib support angular 13 and angular 14.
> 
> @rRaDuCaN @sharmanikhil04 And btw, if @merlosy does not use this account anymore. Do we have any other plans? Do we have any other person can contact with? Or maybe publish this change to anther place?

@winfa  I had upgraded Angular version first to  simulate the error and then added stateChanges . Not sure if we should  stateChanges as Optional without Angular upgrade.
Anyhow, I can see  @mcaden  and @matcho have also committed to this library, So, requesting them to please review this PR and merge this if they have write access . Otherwise, we have to start thinking of other plans

**Comment by winfa** [comment]
> > Hi @sharmanikhil04, I was trying to build based on your PR but found errors when run `npm run build` or `npm run build:lib`. Maybe the angular upgrade part may should be covered by nx.
> > Also, not sure if it is possible to keep the dependency as angular 13, and just update the code and keep stateChanges as optional to make this lib support angular 13 and angular 14.
> > @rRaDuCaN @sharmanikhil04 And btw, if @merlosy does not use this account anymore. Do we have any other plans? Do we have any other person can contact with? Or maybe publish this change to anther place?
> 
> @winfa I had upgraded Angular version first to simulate the error and then added stateChanges . Not sure if we should stateChanges as Optional without Angular upgrade. Anyhow, I can see @mcaden and @matcho have also committed to this library, So, requesting them to please review this PR and merge this if they have write access . Otherwise, we have to start thinking of other plans

Sure. Thank you for the reply.

**Review by joanroig — APPROVED** [review]

**Comment by merlosy** [comment]
Hi all! Sorry for the delay, and thank you for the time and effort in contributing. I have a lot going on atm but I'm not giving up on this.
I'll have a look at this soon!

**Comment by JeremieLapert** [comment]
> Hi all! Sorry for the delay, and thank you for the time and effort in contributing. I have a lot going on atm but I'm not giving up on this. I'll have a look at this soon!


Can we help you maintain the project?

**Review by easyeviction — APPROVED** [review]

**Review by ludonoel1 — APPROVED** [review]

**Review by Soeren86n — APPROVED** [review]

**Review by Leccho — APPROVED** [review]

**Review by Leccho — APPROVED** [review]

**Review by merlosy — APPROVED** [review]

**Pull request merged**